DESIGN.md

```diff
@@ -73,77 +157,225 @@ Flow control support:
+```
+
+### Terminal behavior
+
+A producer can terminate a stream by emitting `onComplete` or `onError`. A consumer can terminate a stream by calling `cancel`/`dispose`.
+
+Any resource cleanup of the source or operators must account for any of these three termination events. In other words, if an operator needs cleanup, then it should register the cleanup callback with `cancel`/`dispose`, `onError` and `onComplete`.
+
+The final `subscribe` will *not* invoke `cancel`/`dispose` after receiving an `onComplete` or `onError`.
+
+### JVM target and source compatibility
+
+The 2.x version will target JDK6+ but the source will be using some of the Java 8 features.
+We'll use the retrolambda project to generate JDK6 Byte-code from the Java 8 sources.
+The only Java 8 features we intend to use are: lambda expressions, method references and try-with-resources statements.
+Default methods may be considered if retrolambda support reach an acceptable level of maturity.
+
+It still up to discussion to know how we will generate multiple targets?
+Options are:
+
+- two artifcat ids: rxjava-jdk6, rxjava-jdk8
+- classifiers: jdk6, jdk8
 
-- If `request(n)` behavior is supported in the stream implementation, then:
- - pull-based creation must support `request(n)` semantics
- - push-based creation must provide a default *onBackpressure* strategy
-- If `request(n)` behavior is not supported in the stream implementation, then:
- - push-based creation can push without consideration of a backpressure strategy
- - pull-based creation should be discouraged
+The intent is to let Android users consume the new version of RxJava.
+(Not sure if JDK6 is the appropriate requirement, JDK7+ is maybe sufficient).
 
-##### Destruction
+##### Custom Observable, Single, Completable, or Flowable (To be confirmed)
 
-A producer can terminate a stream by emitting `onComplete` or `onError`. A consumer can terminate a stream by calling `cancel`.
+An implementation of an Observable which can be subscribed to with an `Observer`. Custom (`Observable`|`Single`|`Completable`|`Flowable`) classes would implement the interface `Consumable<S>` where the generic type `S` is a `Flowable.Subscriber<T>`, `Observable.Observer<T>`, `Completable.Subscriber<T>`, or a `Single.Subscriber<T>` depending on which semantics the custom class will follow. This choice would also affect the kinds of observables the custom class could interop with. For instance, `Flowable#merge` could operate over the standard `Flowable` or any custom class which implements `Consumable<Flowable.Subscriber<T>>`. 
 
-Any resource cleanup of the source or operators must account for any of these three termination events. In other words, if an operator needs cleanup, then it should register the cleanup callback with `cancel`, `onError` and `onComplete`. 
+For more information see the proof of concept project [Consumable](https://github.com/stealthcode/Consumable). 
```

Comment:
> My guess is that the class is aimed at writing generic operators that work with all sorts of Subscribers

No this is not the goal of a core consumable interface. The different types of subscribers each have very specific semantics and this is in no way attempting to bridge them to write operators that function over all subscriber semantics. The purpose of this change is to enable different types of `Flowable`s that can all use the standard flowable operators. Please link to the Consumable proof of concept for more details on why and how.